### PR TITLE
Sales Person Commission Summary and More

### PR DESCRIPTION
**Pull request by deepeshgarg007**
Fix For issue #6935 

1. Incentives for sales person are automatically calculated.
2. Sales Person wise commission summary is also now available.
3. Dashboard For Sales Person is added.
4. Commission rate for Sales Person is also defined.

Screenshots:
screen shot 2018-09-14 at 2 23 36 pm
screen shot 2018-09-14 at 2 27 43 pm

### CODE REVIEW

File: erpnext/selling/report/sales_person_commission_summary/sales_person_commission_summary.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+ "add_total_row": 1, 
+ "creation": "2018-09-11 17:49:27.256304", 
+ "disabled": 0, 
+ "docstatus": 0, 
+ "doctype": "Report", 
+ "idx": 0, 
+ "is_standard": "Yes", 
+ "letter_head": "Gadgets International",
```
Comment (nabinhait) [inline erpnext/selling/report/sales_person_commission_summary/sales_person_commission_summary.json]: remove

File: erpnext/selling/doctype/sales_team/sales_team.json [hunk 2]
```diff
@@ -145,24 +180,26 @@
    "print_hide": 0, 
    "print_hide_if_no_value": 0, 
    "read_only": 0, 
+   "remember_last_selected_value": 0, 
    "report_hide": 0, 
    "reqd": 0, 
    "search_index": 0, 
    "set_only_once": 0, 
+   "translatable": 0,
```
Comment (nabinhait) [inline erpnext/selling/doctype/sales_team/sales_team.json]: Add commission rate field (read only)

File: erpnext/selling/report/sales_person_commission_summary/sales_person_commission_summary.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+def get_columns(filters):
+	if not filters.get("doc_type"):
+		msgprint(_("Please select the document type first"), raise_exception=1)
+
+	return [filters["doc_type"] + ":Link/" + filters["doc_type"] + ":140",
+		_("Customer") + ":Link/Customer:140", _("Territory") + ":Link/Territory:100", _("Posting Date") + ":Date:100",
+		_("Amount") + ":Currency:120",
+		_("Sales Person") + ":Link/Sales Person:140", _("Contribution %") + "::110",
+		_("Contribution Amount") + ":Currency:140",_("Incentives") + ":Currency:140"]
```
Comment (nabinhait) [inline erpnext/selling/report/sales_person_commission_summary/sales_person_commission_summary.py]: Add "Commission Rate" column

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/selling/sales_common.js [hunk 4]
```diff
@@ -179,8 +179,43 @@ erpnext.selling.SellingController = erpnext.TransactionController.extend({
 				sales_person.allocated_percentage / 100.0,
 				precision("allocated_amount", sales_person));
 
-			refresh_field(["allocated_percentage", "allocated_amount"], sales_person.name,
-				sales_person.parentfield);
+				frappe.db.get_value("Sales Person",sales_person.sales_person, "commission_rate")
```
Comment (nabinhait) [inline erpnext/selling/sales_common.js]: Use add_fetch and write a function to calculate incentive_amount and call it from sales_person and incentive_percentage

File: erpnext/setup/doctype/sales_person/sales_person.json [hunk 5]
```diff
@@ -108,11 +111,44 @@
...
+   "in_filter": 0, 
+   "in_global_search": 0, 
+   "in_list_view": 0, 
+   "in_standard_filter": 0, 
+   "label": "Commission Rate", 
+   "length": 0, 
+   "no_copy": 0, 
+   "permlevel": 0, 
+   "precision": "", 
+   "print_hide": 0,
```
Comment (nabinhait) [inline erpnext/setup/doctype/sales_person/sales_person.json]: print_hide =1

File: erpnext/setup/doctype/sales_person/sales_person.py [hunk 6]
```diff
@@ -6,16 +6,35 @@
...
+	def onload(self):
+		self.load_dashboard_info()
+
+	def load_dashboard_info(self):
+		company_default_currency = get_default_currency() 
+
+		allocated_amount = frappe.db.sql("""
+			select sum(allocated_amount)
+			from `tabSales Team`
+			where sales_person = %s and docstatus=1
```
Comment (nabinhait) [inline erpnext/setup/doctype/sales_person/sales_person.py]: parenttype='sales order'

File: erpnext/setup/doctype/sales_person/sales_person.py [hunk 7]
```diff
@@ -35,4 +54,36 @@ def validate_employee_id(self):
...
+
+	out.update(dict(frappe.db.sql('''select
+			unix_timestamp(dt.transaction_date), count(st.parenttype)
+		from
+			`tabSales Order` dt, `tabSales Team` st
+		where
+			st.sales_person = %s and st.parent = dt.name and dt.transaction_date > date_sub(curdate(), interval 1 year) 
+			group by dt.transaction_date ''', name)))
+
+	out.update(dict(frappe.db.sql('''select
```
Comment (nabinhait) [inline erpnext/setup/doctype/sales_person/sales_person.py]: fix overwriting

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/selling/report/sales_person_commission_summary/sales_person_commission_summary.py (hunk 3 above)
Comment (rmehta) [reply erpnext/selling/report/sales_person_commission_summary/sales_person_commission_summary.py]: use the new `dict` style column definitions !

**Comment by rmehta** [comment]
👍  for first pull request!

**Pull request closed without merging**